ipq40xx: uboot envtools fix for GL.iNet GL-B1300 board

barhom opened the pull request:
This pull request makes the uboot env partition readable on the GL.iNet GL-B1300 board.

chunkeey commented on package/boot/uboot-envtools/files/ipq40xx:
```diff
@@ -32,7 +32,8 @@ ubootenv_mtdinfo () {
 openmesh,a42 |\
-openmesh,a62)
+openmesh,a62 |\
+glinet,gl-b1300)
```
please stick to the lexicographical ordering and move the glient,gl-b1300 in front of the openmesh,a42. Thanks.

barhom replied:
Fixed, thanks.

mkresin commented:
> This commit adds the nescessary settings to allow manipulating the uboot environment on the GL.iNet GL-B1300 board.

You commit message should explain why the env need to be manipulated? Usually we don't add support for doing so if not necessarily required (active partition indication or similar).

barhom committed:
ipq40xx: uboot-envtools configuration for GL.iNet GL-B1300.
This commit adds the nescessary settings to allow reading the uboot environment variables on the GL.iNet GL-B1300 board.

barhom commented:
I have changed so that this pull request only allows the uboot environment to be read.

mkresin commented:
> My reason is because I wanted to be able to change the following settings and possibly able to add additional key=value pairs

That is really special purpose stuff and should be done in your local tree instead.

> I have changed so that this pull request only allows the uboot environment to be read.

I still tend to NAK this patch. It is still special purpose stuff and should be either done for all board or for none. And I tend to say it should be done for none...

if no one else picks the PR within the next two weeks, I'll close it as rejected.

mkresin commented:
> This was discussed yesterday on the #openwrt channel and we came to the conclusion that there is no reason not to include the part that can read the uboot environment. We did however agree that uboot should stay read-only which is why I removed one commit.

Considering your use case, the full blown u-boot-envtools package doesn't make much sense. It only makes sense if you need to write to the u-boot env partition. During write the checksum of the env need to be updated.

You might want to do the same as [mtd_get_mac_ascii](https://github.com/openwrt/openwrt/blob/master/package/base-files/files/lib/functions/system.sh#L23) does:

```
$(strings "/dev/mtd5" | sed -n 's/^'"bootcmd"'=//p')
```

barhom commented:
@mkresin 
Isnt the full uboot-envtools package still optional even after this commit? This commit should simply add the necessary configuration for uboot-envtools should you choose to install it.

Escalion commented:
@mkresin I'm siding with @barhom on this one.
A fair few of the ipq40xx targets supports the uci configuration of the u-boot env and adding uboot-envtools is completely optional from the build settings.
It is an ease-of-use enhancement and in my opinion should be merged as at 5acf7c4.

mkresin commented:
Obviously no one else is going to merge the PR. I'm closing the PR for now.
